### ISSUE TITLE
ElasticNet attack uses quadratic learning rate decay instead of square root

### ISSUE THREAD

**Issue by rsokl** [Trusted-AI/adversarial-robustness-toolbox#1829]
https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/8aa8c01b987c60aaca0cff3e4b2af5098218dce2/art/attacks/evasion/elastic_net.py#L193

According to [the paper](https://arxiv.org/pdf/1709.04114.pdf), I believe this should be:

```python
decayed_learning_rate = learn_rate * (1 - global_step / decay_steps) ** 0.5 + end_learning_rate 
```

**Comment by beat-buesser** [comment]
Hi @rsokl Thank you very much for raising this issue! I think this is indeed a bug and your solution is correct. Did you have a chance to test the influence of the change on the attack's performance?

**Comment by rsokl** [comment]
> Did you have a chance to test the influence of the change on the attack's performance?

I hadn't been actually using the ART implementation, I was only referencing the it, so I can't speak to this

**Comment by rsokl** [comment]
Just a quick follow-up, it does look like the change substantially affects the results of the attack at least for the fixed hyperparameters that my colleague was using. In the instance we were looking at, the new lr-scheduler leads to perturbations that are much denser and more intense
